pkg/coveragedb/coveragedb.go

```diff
@@ -81,23 +99,28 @@ func SaveMergeResult(ctx context.Context, client spannerclient.SpannerClient, de
 	ssCache := make(map[string][]string)
 
 	session := uuid.New().String()
-	mutations := []*spanner.Mutation{}
+	var mutations []*spanner.Mutation
 
 	for {
-		var mcr MergedCoverageRecord
-		err := dec.Decode(&mcr)
+		var wr JSONLWrapper
+		err := dec.Decode(&wr)
 		if err == io.EOF {
 			break
 		}
 		if err != nil {
 			return rowsCreated, fmt.Errorf("dec.Decode(MergedCoverageRecord): %w", err)
 		}
-		if mcr.FileData == nil {
-			return rowsCreated, errors.New("field MergedCoverageRecord.FileData can't be nil")
+		if wr.MCR == nil && wr.FL == nil {
+			return rowsCreated, errors.New("JSONLWrapper can't be empty")
+		}
+		if mcr := wr.MCR; mcr != nil {
```

Comment:
Nit:

```
if mcr := wr.MCR; mcr != nil {
} else if fl := ... ; fl != nil {
} else {
  return rowsCreated, errors.New("... can't be empty")
}

Reply:
We'll miss data in case of the interface misuse (think about both MCR and FL initialized).
With current code we'll just silently accept both sections.

And I find validation here too defensive...

Reply:
Now done.